### PR TITLE
Better Testing with new typo marking 

### PR DESCRIPTION
**Pull request by caffodian**

### CODE REVIEW

File: bender.js [hunk 1]
```diff
@@ -48,7 +48,8 @@ var config = {
             basePath: 'tests/',
             paths: [
                 'smoke/**',
-				'typowalking/**',
+				        'typowalking/**',
```
Comment (caffodian) [inline bender.js:51]: very nitpicky, but what's with this random indent?

File: plugins/nanospell/plugin.js [hunk 2]
```diff
@@ -225,6 +225,18 @@
 				},
 				editorFocus: true
 			});
+			editor.addCommand('nanospellReset', {
+				exec: function(editor) {
+					spellcache = [];
+					suggestionscache = [];
+					ignorecache = [];
```
Comment (caffodian) [inline plugins/nanospell/plugin.js:232]: something I just noticed, we instantiate these as arrays, then we use them like objects.  not sure if we actually benefit from this quirk

File: tests/spellchecking/spellcheck.js [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+			this.server.respondImmediately = true;
+
+			// for these tests we don't really care that much about the
+			// mock data, just that it returns something vaguely resembling the server call
+			var suggestions = {
+				"result": {
+					"appkes": ["apples"],
+					"pearrs": ["pears"],
+					"bannanas": ["bananas"],
+          "missspelling": ["misspelling"]
```
Comment (caffodian) [inline tests/spellchecking/spellcheck.js]: weird indent again

File: tests/spellchecking/spellcheck.js [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+				editor = bot.editor,
+				resumeAfter = bender.tools.resumeAfter,
+				starterHtml = '<p>Paragraph with missspelling</p>';
+
+			bot.setHtmlWithSelection(starterHtml);
+
+			resumeAfter(editor, 'spellCheckComplete', function() {
+				var paragraph = editor.editable().findOne('p');
+
+				tc.assertHtml('<p>Paragraph with <span class="nanospell-typo">missspelling</span></p>', paragraph.getOuterHtml());
```
Comment (caffodian) [inline tests/spellchecking/spellcheck.js:54]: This doesn't have to be done now,

But I think we could have used a utility function here:

```
getMarkedWords: function (editor) {
     ... return [contents of all span.nanospell-typo]
}
```

Then use the array matching assertion.   Unless we care specifically about the details of the html, which we might.  I can't think of a reason, though.

File: tests/spellchecking/spellcheck.js [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+      );
+
+      resumeAfter(editor, 'spellCheckComplete', function() {
+        var paragraph = editor.editable().findOne('p');
+
+        tc.assertHtml(
+          '<p><span class="nanospell-typo">appkes</span><br>' +
+            '<span class="nanospell-typo">pearrs</span><br>' +
+            '<span class="nanospell-typo">bannanas</span>' +
+          '</p>',
```
Comment (caffodian) [inline tests/spellchecking/spellcheck.js]: going back to my previous comment, maybe we do care about the html here (breaks are not actually wrapped in spans by accident, etc)

Comment (sbaughman) [reply tests/spellchecking/spellcheck.js]: yeah it seemed to me that there was a need for tests to make sure that our output html exactly matches what we expect

**Pull request merged**